doi-transformer/src/test/java/no/unit/nva/doi/transformer/CrossRefConverterTest.java

```diff
@@ -474,62 +497,191 @@ public void toPublicationPreservesOrcidWhenAuthorHasOrcid() throws InvalidIssnEx
+        String actualPublisher = ((Book)publicationContext).getPublisher();
+        assertThat(actualPublisher, is(equalTo(SAMPLE_PUBLISHER)));
+    }
 
-        List<Contributor> contributors = actualDocument.getEntityDescription().getContributors();
-        List<List<Organization>> organisations =  contributors.stream()
-                .map(Contributor::getAffiliations)
-                .filter(Objects::nonNull)
-                .collect(Collectors.toList());
+    @Test
+    @DisplayName("toPublication sets PeerReviewed to True PublicationContext when CrossrefDocument has Reviewer")
+    public void toPublicationSetsPeerReviewedInPublicationContextWhenCrossrefDocumentHasReviewer()
+            throws InvalidIssnException, InvalidIsbnException {
+        CrossRefDocument crossRefDocument = createSampleDocumentBook();
+        CrossrefReview crossrefReview = new CrossrefReview();
+        crossrefReview.setRecommendation(REVIEW_RECOMMENDATION);
+        crossRefDocument.setReview(crossrefReview);
+        Publication actualDocument = toPublication(crossRefDocument);
+        var publicationContext =  actualDocument.getEntityDescription().getReference().getPublicationContext();
+        assertTrue(((Book)publicationContext).isPeerReviewed());
+    }
 
-        assertTrue(organisations.isEmpty());
+    @Test
+    @DisplayName("toPublication sets Url in PublicationContext when CrossrefDocument has link")
+    public void toPublicationSetsUrlInPublicationContextWhenCrossrefDocumentHasLink()
+            throws InvalidIssnException, InvalidIsbnException {
+        CrossRefDocument crossRefDocument = createSampleDocumentBook();
+        crossRefDocument.setLink(List.of(createLink()));
+        Publication actualDocument = toPublication(crossRefDocument);
+        var publicationContext =  actualDocument.getEntityDescription().getReference().getPublicationContext();
+        var actualLink = ((Book) publicationContext).getUrl();
+        assertThat(actualLink.toString(), is(equalTo(SAMPLE_LINK)));
+    }
+
+
+    @Test
+    @DisplayName("toPublication handles all interesting and required fields in CrossrefDocument for book")
+    public void toPublicationHandlesAllInterestingAndRequiredFieldsInCrossrefDocumentForBook()
+            throws InvalidIssnException, InvalidIsbnException {
+        CrossRefDocument crossRefDocument = createSampleDocumentBook();
+        CrossrefDate crossrefDate = createCrossrefDate();
+
+
+        crossRefDocument.setPublisher(SAMPLE_PUBLISHER);
+        crossRefDocument.setTitle(List.of(SAMPLE_DOCUMENT_TITLE));
+        crossRefDocument.setDoi(SOME_DOI);
+        crossRefDocument.setUrl(SOME_DOI_AS_URL.toString());
+        crossRefDocument.setLink(List.of(createLink()));
+        crossRefDocument.setCreated(crossrefDate);
+        crossRefDocument.setDeposited(crossrefDate);
+        crossRefDocument.setIssued(crossrefDate);
+
```

Comment:
assertThat(crossRefDocument,doesNotValueEmptyValues())

Reply:
Move  the assertions out to its own method?

Reply:
yes